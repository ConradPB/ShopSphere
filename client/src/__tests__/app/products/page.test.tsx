import { render, screen } from "@testing-library/react";
import ProductsPage from "@/app/products/page";

jest.mock("@/components/ProductGrid", () =>
  jest.fn(() => <div>Mocked Grid</div>)
);

describe("ProductsPage", () => {
  it("renders the products page content", async () => {
    await act(async () => {
      render(<ProductsPage />);
    });

    // Updated to match your actual text
    expect(screen.getByText(/Explore Our Products/i)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<ProductsPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
