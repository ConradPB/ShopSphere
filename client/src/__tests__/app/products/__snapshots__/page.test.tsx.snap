// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductsPage matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="relative min-h-screen bg-gradient-to-b from-neutral-950 via-slate-900 to-neutral-950 text-white py-20"
  >
    <div
      class="absolute inset-0 pointer-events-none overflow-hidden"
    >
      <div
        class="absolute -top-48 left-1/2 -translate-x-1/2 w-[600px] h-[600px] bg-cyan-500/10 blur-[130px] rounded-full"
      />
      <div
        class="absolute bottom-0 right-1/4 w-[400px] h-[400px] bg-purple-500/10 blur-[110px] rounded-full"
      />
    </div>
    <div
      class="max-w-7xl mx-auto px-6 relative z-10"
    >
      <div
        style="opacity: 0; transform: translateY(30px);"
      >
        <h1
          class="text-4xl md:text-5xl font-extrabold text-center mb-14 font-display tracking-tight"
        >
          Explore Our Products
        </h1>
      </div>
      <p
        class="text-center py-20 text-gray-400 animate-pulse"
      >
        Loading products...
      </p>
    </div>
    <div
      class="pointer-events-none absolute inset-x-0 top-0 h-40 bg-gradient-to-b from-neutral-950 via-transparent to-transparent"
    />
    <div
      class="pointer-events-none absolute inset-x-0 bottom-0 h-40 bg-gradient-to-t from-neutral-950 via-transparent to-transparent"
    />
  </main>
</DocumentFragment>
`;
